### ISSUE TITLE
How to use with tensorflow estimator API?

### ISSUE THREAD

**Issue by ishwara-bhat** [InFoCusp/tf_cnnvis#67]
Hi,
This is not really an issue, but a query. I went through the example and understood that after session is run, we should # activation visualization and then # deconv visualization# decon. 

i.e. the two steps are to be inserted between session run and session close. 

I am using estimator API and its flow is not the same. In my prediction code, I have 
model = tf.estimator.Estimator(model_function, model_dir)
then 
input_function preparation (with lambda function).

Then I run prediction
classes = model.prediction(input_function)

Any suggestions on how to add the two steps to the estimator API code flow?

Thanks
Ishwar

**Comment by falaktheoptimist** [comment]
If you have the model graph with weights already present, you could just pass in that in the input. Or you could save out model to disk, load it and pass that into tf_cnnvis API. 
I'm guessing you would have to add a placeholder for inputs externally which pass on the input to the first layer in your model since the model would not have that owing to usage of the estimator API. You can then pass in the inputs into the model via this placeholder as demonstrated in the examples.
Let us know if you hit any issues in doing this.

**Comment by ishwara-bhat** [comment]
Thanks. It works. 
Here is the code for restoring TF savedmodel (.meta model file i.e. the set of graph.txt,checkpoint.data-0000-of-0001, checkpoint.index) and using InFoCusp/tf_cnnvis. 
tf.reset_default_graph()
session = tf.session()
input_placeholder = tf.placeholder(tf.float32, shape=[1,224,224,3] #shape of images in sample image
saver = tf.train.import_meta_graph('your meta file name.meta')
save.restore(session, tf.train.latest_checkpoint(model_dir)
im = np.expand_dims( ... axis=0)
layers = ['r','p','c']
is_success= activation_visualization(sess_graph_path=session,value_feed_dict={input_placeholder:im},layers=layers,......)

Great tool. Thanks!
I still have observations: imresize , imread get tf 1.10 warning of deprecation. 
I still have some type error related to mappers. But the basic call works and filter images get generated in ouput folders.
